### ISSUE TITLE
Template options

### ISSUE THREAD

**Issue by wesselvdv** [theory/sqitch#369]
What I am looking for is a way to define the naming convention and location of all the scripts that are generated by Sqitch.

I am currently using software from IntelliJ, and they have a nice feature called file nesting. The [feature](https://www.jetbrains.com/help/webstorm/file-nesting-dialog.html) is described nicely on their site. It basically allows you to stay sane, by grouping files that are somehow related. (like revert, deploy, test and verify scripts) It's however not supported when these files are in fact in different directories like with Sqitch.

So instead of the following structure:

#### Project tree
 * deploy
   * schema
     * change.sql
 * revert
   * schema
     * change.sql
 * verify
   * schema
     * change.sql

I would love to be able to define the following (which would work nicely with file nesting):

#### Project tree
 * schema
   * tables
      * change.deploy.sql
      * change.revert.sql
      * change.verify.sql

**Comment by wesselvdv** [comment]
@theory I am not saying the directory route is a bad approach. Only offering something along the lines of what I am suggesting would push people towards products that offer file nesting like functionalities or they'd probably go mad. 

Sidetrack:
* Are those options that you're mentioning now already available for configuration?
* What is the best place to look for all the different configuration options? If it's the lib directory in the repo, then I am already looking in the right place.

**Comment by theory** [comment]
Yes, they are documented in [the config docs](/theory/sqitch/blob/master/lib/sqitch-config.pod), but the [sqitch-configuration](https://github.com/theory/sqitch/blob/master/lib/sqitch-configuration.pod) docs go into detail on hierarchical target/engine/core config.

**Comment by wesselvdv** [comment]
Is there anyway within the current capabilities of Sqitch that I can do to mimic something like this? I was thinking of something with symlinks, but that will definitely get very complex I am afraid.

**Comment by theory** [comment]
Afraid not, no. I marked this as a feature request in case someone wants to work on it, though.
